internal/cloud/realm/user.go

```diff
@@ -92,11 +93,139 @@ type User struct {
 // UserIdentity is a Realm application user identity
 type UserIdentity struct {
 	UID          string                 `json:"id"`
-	ProviderType string                 `json:"provider_type"`
+	ProviderType ProviderType           `json:"provider_type"`
 	ProviderID   primitive.ObjectID     `json:"provider_id"`
 	ProviderData map[string]interface{} `json:"provider_data,omitempty"`
 }
 
+// ProviderType is a Realm application provider type
```

Comment:
[nit/style] prefer to use "Realm app" when bringing up apps.  this was newly declared, so feel free to drive-by/update any incorrect instances you find (no need to seek them out though)

also, just within this package, let's rename this (and associated `const`s) to `AuthProviderType`